src/FlaUILibrary/keywords/mouse.py

```diff
@@ -212,3 +212,73 @@ def drag_and_drop(self, start_identifier, end_identifier, msg=None):
         module.action(Mouse.Action.DRAG_AND_DROP,
                       Mouse.create_value_container(element=start_element, second_element=end_element),
                       msg)
+
+    @keyword
+    def click_open(self, click_element_identifier: str, open_element_identifier: str, 
+                    focus_element_identifier_before_click: str = None, focus_element_identifier_after_open: str = None , 
+                    max_repeat: int = 5, timeout_between_repeates: int = 1000, ignore_if_already_open: bool = True, msg=None):
+        """
+        Left clicks to element by an XPath and excepts an element to be opened.
+        It repeats the act of clicking until the excepted element is there by waiting for given seconds each time.
+        Maximum repeating time could also be changed
+
+        XPaths syntax is explained in `XPath locator`.
+
+        If element could not be found by xpath an error message will be thrown.
+
+        Arguments:
+        | Argument   | Type   | Description                   |
+        | click_element_identifier | string | XPath identifier from element to be clicked |
+        | open_element_identifier | string | XPath identifier from element to be opened after clicking click_element |
+        | focus_element_identifier_before_click | string | XPath identifier from element to be focused before clicking click_element |
+        | focus_element_identifier_after_open | string | XPath identifier from element to be focused after openning open_element |
+        | max_repeat | int | Maximum number of tries |
+        | timeout_between_repeates | int | wait time in milli seconds in between every try |
+        | ignore_if_already_open | bool | the keyword will not be executed if excepted element is already open |
+        | msg | string | Custom error message |
+
+        Examples:
+        | Click Open  <XPATH>  <XPATH>  |
+
+        """
+        module = self._container.create_or_get_module()
+        module.action(Mouse.Action.LEFT_CLICK_OPEN,

```

Comment:
Here you should get all elements by module usage and send as argument parameter

Reply:
I know what you mean, This was my first idea also so i follow the same footsteps of the other mouse keyword
The problem is the seceond element which i am intending is not there if i try to get the element instead of xpath in this keyord python module then it will fail. Because it will exist after the clicking has done 
And most probably the focus element too. Which means the only element i could convert here will be the clicking element itself. All others should remain xpahts